In [1]:
from math import pi
import sympy
from sympy import *

p= symbols('p')
n, v, m, k, T, d, rho = symbols('n v m k T d, rho', real='True', positive='True')

Question 1 Part (a):
$$\langle v \rangle = \frac{\int_{-\infty}^{\infty} n(v) v\, d^{3}v}{\int_{-\infty}{\infty} n(v)\, d^{3}v} = \frac{1}{n}\int_{-\infty}^{\infty} n(v) v\, d^{3}v$$
Assuming non-relativistic motion, 
$$p = v m$$
$$n(v) d^{3}v \rightarrow \frac{4 \pi v^2 n}{(2 \pi m k_b T) ^ {3/2}} e^{(-m v^2)/(2 k_b T)}\, dv$$
Thus,
$$\langle v \rangle = \frac{1}{n} \int_{0}^{\infty} \frac{4 \pi v^3 n}{(2 \pi m k_b T) ^ {3/2}} e^{(-m v^2)/(2 k_b T)}\, dv$$


In [2]:
average = ((4 * pi * v**3 * n)/((2 * pi * m * k * T)**Rational(3,2))) * exp((-m * v**2)/(2 * k * T))
integrate(average / n,(v,0, oo))

2*sqrt(2)*sqrt(T)*sqrt(k)/(sqrt(pi)*m**(7/2))

$$\langle v \rangle = \frac{2 \sqrt{2} \sqrt{T} \sqrt{k_b}}{\sqrt{\pi} m^{7/2}}$$

Question 1 Part (b):
$$P = \frac{1}{3} \int n(p)\, v\, p\, d^{3}p$$
$$n(p) \, d^{3}p \rightarrow \frac{4 \pi n}{(2 \pi m k_{b} T)^{3/2}} p^{2} e^{-p^2 / (2 m k_{b} T)}\, dp$$
We are also assuming non-relativistic motion, and as such,
$$v = \frac{p}{m}$$
Putting this all together,
$$P = \frac{1}{3} \int_{0}^{\infty} \frac{4 \pi n}{(2 \pi m k_{b} T)^{3/2}} e^{-p^2 / (2 m k_{b} T)} \frac{p^4}{m}\, dp$$

In [3]:
inner = 4*pi*n *(2*pi*m*k*T)**-Rational(3, 2) * exp(-p**2/(2*m*k*T)) * ((p**4) / m)
integrate(inner / 3, (p, 0, oo))

T*k*n

Therefore,
$$P = n k_{b} T$$

Question 1c:
The specific internal energy in integral form can be formulated as,
$$u = \frac{1}{\rho} \int_{0}^{\infty} n(p) \varepsilon (p)\, dp$$
where $\varepsilon (p)$ is the energy of a particle with momentum $p$,
$$\varepsilon (p) = mc^2 (\sqrt{1 + \frac{p^2}{m^2 c^2}} - 1)$$
Taking the first term of the Taylor series expansion, under the conditions referring to non-relativistic motion,
$$\varepsilon (p) = \frac{p^2}{2m}$$
The expanded specific internal energy integral can be written as,
$$u = \frac{1}{\rho} \int_{0}^{\infty} \frac{4 \pi n}{(2 \pi m k_{b} T)^{3/2}} e^{-p^2 / (2 m k_{b} T)} \frac{p^4}{2m}\, dp$$

In [4]:
inner2 = 4*pi*n *(2*pi*m*k*T)**-Rational(3, 2) * exp(-p**2/(2*m*k*T)) * ((p**4) / (2*m))
integrate(inner2 / d, (p, 0, oo))

3*T*k*n/(2*d)

Therefore,
$$u = \frac{3 n k_{b} T}{2 \rho} $$

Question 2 Part (a):
$$n_{i} = \frac{X_{i} m_{H} \rho}{A_{i}}; \, n_{H} = \frac{X_{H} m_{H} \rho}{A_{H}}$$
$$\frac{n_i}{n_H} = \frac{X_i A_H}{X_H A_i}$$
The abundance of elements in the Sun is given by,
$$\mathcal{A_i} = 12 + \log_{10} (\frac{n_i}{n_H})$$
We can rewrite the abundance formula as,
$$10^{\mathcal{A_i} - 12} = \frac{X_i A_H}{X_H A_i}$$

Let $X_{H} = 0.707$, we can then calculate;
$$X_{He} = 0.707 * 4 * 10^{10.99 -12}$$

In [5]:
simplify(10**(10.99-12) * 4 * 0.707)

0.276362686086303

$$X_{He} = 0.276$$

In [6]:
simplify(10**(8.93-12) * 16 * 0.707)

0.00962807348814528

$$X_{O} = 0.010$$

In [7]:
simplify(10**(8.56-12) * 12 * 0.707)

0.00308035421666954

$$X_{C} = 0.003$$

In [8]:
simplify(10**(8.09-12) * 20 * 0.707)

0.00173960004192871

$$X_{Ne} = 0.002$$

In [9]:
simplify(10**(8.05-12) * 14 * 0.707)

0.00111057386606809

$$X_{N} = 0.001$$

In [10]:
simplify(10**(7.58-12) * 24 * 0.707)

0.000645105367676728

$$X_{Mg} = 0.0006$$

In [11]:
simplify(10**(7.55-12) * 28 * 0.707)

0.000702388585326786

$$X_{Si} = 0.0007$$

In [12]:
simplify(10**(7.54-12) * 56 * 0.707)

0.00137280053831166

$$X_{Fe} = 0.001$$

In [13]:
simplify(10**(7.21-12) * 32 * 0.707)

0.000366918316426484

$$X_{S} = 0.0004$$

Question 2 Part (b):
$$\frac{1}{\mu_{I}} = X + \frac{Y}{4} + \frac{Z}{\langle \mathcal{A} \rangle_{metals}}$$
where $\langle \mathcal{A} \rangle_{metals}$ is the mean atomic mass of the metals, which in the Sun is $\approx$ 20.
Since $X = 0.707$, $Y = 0.276$ and $Z = 0.017$,
$$\mu_{I} = \frac{1}{0.707 + (0.276 / 4) + (0.017 / 20)}$$

In [14]:
simplify(1 / (0.707 + (0.276/4) + (0.017/20)))

1.28724979082191

$$\mu_{I} = 1.29$$

$$\frac{1}{\mu_{e}} = X + \frac{Y}{2} + Z\langle \frac{\mathcal{Z}}{\mathcal{A}} \rangle_{metals}$$
where $\langle \frac{\mathcal{Z}}{\mathcal{A}} \rangle_{metals}$ represents the ratio of protons averaged over all metal atoms, which for the Sun is $\approx$ $\frac{1}{2}$.
Therefore we can estimate,
$$\frac{1}{\mu_{e}} \approx \frac{1}{2}(X + 1)$$

In [15]:
simplify(1 / (0.5 * (0.707 + 1)))

1.17164616285882

$$\mu_{e} = 1.17$$

We also know that,
$$\frac{1}{\mu} = \frac{1}{\mu_{I}} + \frac{1}{\mu_{e}}$$
$$\mu = 0.6135$$

Question 3 Part (a):
For an Ideal Gas, electron pressure can be formulated as,
$$P_{e} = \frac{\mathcal{R}}{\mu_{e}} \rho T$$
For a degenerate gas, electron degeneracy pressure is,
$$P_{e, deg} = \frac{h^2}{20 m_{e}} (\frac{3}{\pi})^{2/3} \frac{1}{m_{H}\, ^{5/3}} (\frac{\rho}{\mu_{e}})^{5/3}$$
We can also see that since the star has no hydrogen or helium present, 
$$\mu_{e} \approx \frac{1}{2}(1 + X) \rightarrow \mu_{e} \approx 2$$
Plugging in and then simplifying we have,
$$P_{e} = 4.157 \times 10^{10} \rho$$
$$P_{e, deg} = 1.00315 \times 10^{7} (\frac{\rho}{2})^{5/3}$$
$$4.157 \times 10^{10} \rho = 1.00315 \times 10^{7} (\frac{\rho}{2})^{5/3}$$
$$7.11605 \times 10^{10} \rho^{3} =\frac{\rho^{5}}{32}$$
Thus,
$$\rho = 1.50902 \times 10^6 \frac{kg}{m^3}$$

Question 4 Part (a):
$$e_{gas} = \frac{3 \beta P}{2 \rho}; \, e_{\gamma} = 3 (1 - \beta) \frac{P}{\rho}$$

Question 4 Part (b):
$$U = \frac{3}{2} \int (2- \beta) \frac{P}{\rho}\, dm$$

Question 4 Part (c):
$$\frac{P}{\rho} = \frac{2}{3(2-\beta)}(u_{gas} + u_{\gamma}$$
$$U = -3 \int \frac{P}{\rho}\, dm = - \frac{2}{(2 - \beta)} \int (u_{gas} + u_{\gamma})\, dm = - \frac{2}{(2 - \beta)} U$$